File: implementations/go/summation/summation_large.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const N = 1000000

var arr [N]int64

func setup() {
	for i := 0; i < N; i++ {
		arr[i] = int64(i + 1)
	}
}

func summation() int64 {
	var sum int64 = 0
	for i := 0; i < N; i++ {
		sum += arr[i]
	}
	return sum
}

func main() {
	setup()

	fmt.Println("ready")

	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		fmt.Println(summation())
	}
}

File: implementations/javascript/summation/summation_large.js

const readline = require('readline');

const N = 1000000;
const arr = new Array(N).fill(0);

function setup() {
    for (let i = 0; i < N; i++) {
        arr[i] = i + 1;
    }
}

function summation() {
    let sum = 0;
    for (let i = 0; i < N; i++) {
        sum += arr[i];
    }
    return sum;
}

setup();
process.stdout.write("ready\n");

const rl = readline.createInterface({ input: process.stdin });
rl.on('line', () => {
    process.stdout.write(String(summation()) + "\n");
});